### PR TITLE
Migrate component to TS: Input

### LINKED ISSUE #19122: Migrate components to TS: Input
### Description
Migrate `Input` component to TypeScript

### Technical Details

- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` (name and keys)
- Ensure enums use singular not plural e.g. `Size` not `Sizes`
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation

### Acceptance Criteria

- Component has been converted to TypeScript
- Stories, documentation and tests have been updated inside of the `component-name` folder
- Before/After screen cast of component stories in storybook to show no visual regression after migration

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- This PR fixes #19122 

### Before

https://github.com/MetaMask/metamask-extension/assets/79097544/19de19f1-0917-4a7f-b4ea-401cfc63c9cb

### After

https://github.com/MetaMask/metamask-extension/assets/79097544/c1c27c04-9582-49d1-957d-7e69c2a85ebd

No instances of `INPUT_TYPES`

Screenshot 2023-07-24 at 3 36 57 PM

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/input/input.test.tsx
Comment (dhruvv173) [inline ui/components/component-library/input/input.test.tsx:1]: unit tests pass even though there are a few errors in the test file
image

File: ui/components/component-library/input/input.types.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  /**
+   * The input value, required for a controlled component.
+   */
+  value?: string | number;
+}
+
+export type InputProps<C extends React.ElementType> =
+  PolymorphicComponentPropWithRef<C, InputStyleProps>;
+
+export type InputComponent = <C extends React.ElementType = 'input'>(
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/input/input.types.ts:104]: This looks correct! Really nice work @dhruvv173 I'll do a full review in the coming days. I'm a bit stacked at the moment so forgive me if there is a delay on reviews

[the remark this answers is not in the document]
Comment (dhruvv173) [reply ui/components/component-library/input/input.types.ts:104]: i will wait, no worries. Thank you for the update, appreciate it!

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Really nice work 💯. I made some small adjustments and rebased from `develop`. Thanks for your contribution @dhruvv173

**Review by garrettbear — APPROVED** [review]
👏 @dhruvv173

**Pull request merged**